src/mongocxx/result/update.hpp

```diff
@@ -53,6 +53,12 @@ class MONGOCXX_API update {
     /// @return The number of documents that were modified.
     std::int32_t modified_count() const;
 
+    ///
+    /// Gets the number of documents that were upserted during this operation.
+    ///
+    /// @return The number of documents that were upserted.
+    std::int32_t upserted_count() const;
```

Comment:
No change requested. Was a `std::int64_t` return type considered?

`std::int32_t` is consistent with `result::bulk_write` and the other fields in this class. The CRUD spec uses `Int64`. I think the proposed `std::int32_t` seems like less surprising API.

Reply:
> Was a `std::int64_t` return type considered?

It was not considered. `std::int32_t` was chosen simply because it is the type being returned by `bulk_write::upserted_count()`. If `Int64` needs to be supported, more interfaces than just that of `update` will likely need to be modified.